File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    s = list(s)
    stack = []

    for i in s:
        if i == "(":
            stack.append(i)
        else:
            if not stack:
                return False
            stack.pop()
    
    if len(stack) > 0:
        return False
    else:
        return True

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.swift
func solution(_ s:String) -> Bool
{
    var stack = ""
    
    for i in s {
        if i == "(" {
            stack.append(i)
        } else {
            if stack.isEmpty {
                return false
            }
            stack.popLast()
        }
    }
    return stack.isEmpty
}